namespaces/resource-namespace.md

```diff
@@ -0,0 +1,328 @@
+
+### Data Structure (MSON Element)
+
+Data structure definition using MSON namespace elements.
+
+#### Properties
+
+- `element`: dataStructure (string, fixed)
+
+### Asset (Element)
+
+Arbitrary data asset.
+
+#### Properties
+
+- `element`: asset (string, fixed)
+- `meta`
+    - `class` (array, fixed)
+        - (enum[string])
+            - messageBody - Asset is an example of message-body
+            - messageBodySchema - Asset is an schema for message-body
+- `attributes` (object)
+    - `contentType` (string) - Optional media type of the asset
+    - `href` (Href) - Link to the asset
+- `content` (string) - A textual representation of the asset
+
+## Resource Elements
+
+### Resource (Element)
+
+The Resource representation with its available transitions and attributes.
+
+#### Properties
+
+- `element`: resource (string, fixed)
+- `attributes` (object)
+    - `href` (Templated Href) - URI Template of this resource.
+    - `hrefVariables` (Href Variables) - Definition of URI Template variables used in the `href` property.
+- `content` (array)
+    - (Transition) - State transition available for this resource.
+
+        The `content` MAY include multiple `Transition` elements.
+
+    - (Data Structure) - Data structure representing the resource.
+
+        The `content` MUST NOT include more than one `Data Structure`.
+
+#### Example
+
+```json
+{
+    "element": "resource",
+    "meta": {
+        "title": "Question",
+        "description": "A Question object has the following attributes."
+    },
+    "attributes": {
+        "href": "/questions/{question_id}",
+        "hrefVariables": {
+            "element": "hrefVariables",
+            "attributes": {
+                "name": "question_id"
+            },
+            "content": null
+        }
+    },
+    "content": [
+        {
+            "element": "dataStructure"
+        }
+    ]
+}
+```
+
+### Transition (Element)
+
+A transition is an available progression from one state to another state.
+Exercising a transition initiates a transaction.
+
+The content of this element is array of protocol-specific transactions.
+
+Note: At the moment only the HTTP protocol is supported.
+
+#### Properties
+
+- `element`: transition (string, fixed)
+- `attributes` (object)
+    - `relation` - (string) - Link relation type as specified in [RFC 5988][].
+
+        The value of `relation` attribute SHOULD be interpreted as a link relation
+        between transition's parent resource and the transition's target resource
+        as specified in the `href` attribute.
+
+    - `href` (Templated Href) - The URI template for this transition.
+
+        If present, the value of the `href` attribute SHOULD be used to resolve
+        target URI of the transition.
+
+        If not set, the parent `resource` element `href` attribute SHOULD be
+        used to resolve the target URI of the transition.
+
+    - `parameters` (Href Variables) - Input parameters.
```

Comment:
Yes. I guess we should wait and see for adding the `parameters` at resource level.